### PR TITLE
Add listeners and refactor Android module

### PR DESCRIPTION
**Pull request by filiphosko**

### CODE REVIEW

File: android/src/main/java/com/younics/reachability/Reachability.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+/* TODO: refactor -> one-time configuration into constructor */
+class Reachability {
+    public void isReachable(String hostname, int port, int timeout, final Promise promise) {
+        int tm = timeout > 0 ? timeout : 5000;
+        String hn = hostname != null && !hostname.isEmpty() ? hostname : "8.8.8.8";
+        int pt = port > 0 ? port : 443;
+
+        try {
+            try (Socket soc = new Socket()) {
+                soc.connect(new InetSocketAddress(InetAddress.getByName(hn), pt), tm);
```
Comment (Wreit) [inline android/src/main/java/com/younics/reachability/Reachability.java:20]: Socket should be closed after successful connection "soc.close();"

File: android/src/main/java/com/younics/reachability/ReachabilityAsyncTask.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+        this.isRunning = true;
+    }
+
+    @Override
+    public Boolean doInBackground(Void... records) {
+        while (isRunning) {
+            try {
+                try (Socket soc = new Socket()) {
+                    soc.connect(new InetSocketAddress(InetAddress.getByName(hostname), port), timeout);
```
Comment (Wreit) [inline android/src/main/java/com/younics/reachability/ReachabilityAsyncTask.java:41]: Same as before "soc.close();"

**Review by Wreit — CHANGES_REQUESTED** [review]
Close sockets if you not using them again, besides everything else seems to be okay

File: android/src/main/java/com/younics/reachability/Reachability.java (hunk 1 above)
Comment (filiphosko) [reply android/src/main/java/com/younics/reachability/Reachability.java:20]: @Wreit implemented in https://github.com/Younics/react-native-reachability/pull/4/commits/ba9ac1a0abceb66149e3a1059d3ae0a11ac49963

File: android/src/main/java/com/younics/reachability/ReachabilityAsyncTask.java (hunk 2 above)
Comment (filiphosko) [reply android/src/main/java/com/younics/reachability/ReachabilityAsyncTask.java:41]: @Wreit implemented in https://github.com/Younics/react-native-reachability/pull/4/commits/ba9ac1a0abceb66149e3a1059d3ae0a11ac49963

**Review by Wreit — APPROVED** [review]

**Pull request merged**